### PR TITLE
BUG: GEE fit_history

### PR DESCRIPTION
**Pull request by kshedden**
We can't delete fit_history after fitting the model (while transferring it to the results), because we may need to call update_params or other methods on the parent model that write to the fit history.

Also add some warnings.

### CODE REVIEW

**Comment by kshedden** [comment]
Also closes #5786

File: statsmodels/genmod/tests/test_gee.py [hunk 1]
```diff
@@ -635,14 +635,19 @@ def test_compare_score_test(self, cov_struct):
...
-        mod = gee.GEE(endog, exog, group, cov_struct=cov_struct())
-        score_results = mod.compare_score_test(res_sub)
-        assert_almost_equal(score_results["statistic"],
-            mod_lr.score_test_results["statistic"])
-        assert_almost_equal(score_results["p-value"],
-            mod_lr.score_test_results["p-value"])
-        assert_almost_equal(score_results["df"],
-            mod_lr.score_test_results["df"])
+
+        for f in False, True:
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_gee.py]: Informative variable name for f, brackets around list?

File: statsmodels/genmod/tests/test_gee.py [hunk 2]
```diff
@@ -682,6 +687,14 @@ def test_compare_score_test_warnings(self):
             mod = gee.GEE(endog, exog, group)
             _ = mod.compare_score_test(res_sub)
 
+        # Parent and submodel are the same dimension
+        with assert_warns(UserWarning):
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_gee.py]: Pytest.warns, match kwarg

File: statsmodels/genmod/tests/test_glm.py [hunk 3]
```diff
@@ -1989,7 +1989,7 @@ def test_tweedie_EQL_upper_limit():
         # Un-regularized fit using gradients not IRLS
         fam = sm.families.Tweedie(var_power=2, eql=True)
         model1 = sm.GLM(y, x, family=fam)
-        result1 = model1.fit(method="newton")
+        result1 = model1.fit(method="newton", scale=scale)
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_glm.py]: Is this related to the rest of the Pr?

File: statsmodels/genmod/generalized_estimating_equations.py [hunk 4]
```diff
@@ -1911,11 +1908,21 @@ def qic(self, scale=None):
...
+        model under consideration.
+
+        If the scale parameter is not provided, the estimated scale
+        parameter is used.  Doing this does not allow comparisons of
+        QIC values between models.
         """
 
+        # It is easy to forget to set the scale parameter.  Sometimes
+        # this is intentional, so we warn.
+        if isinstance(self.family, families.Gaussian) and scale is None:
```
Comment (josef-pkt) [inline statsmodels/genmod/generalized_estimating_equations.py]: GEE also allows other families with scale != 1

https://www.statsmodels.org/dev/_modules/statsmodels/genmod/generalized_estimating_equations.html#GEE.estimate_scale
lists the families with scale=1, instead of the complement set

in general there should be a scale option as in GLM, e.g. to handle purely overdispersed Poisson (i.i.d. with scale!=1).

File: statsmodels/genmod/tests/test_glm.py (hunk 3 above)
Comment (jbrockmendel) [reply statsmodels/genmod/tests/test_glm.py]: before this fades into distant memory, any thoughts here?

File: statsmodels/genmod/tests/test_gee.py [hunk 5]
```diff
@@ -1883,8 +1896,10 @@ def test_ql_known(family):
     assert_allclose(ql1, qle1[0], rtol=1e-4)
     assert_allclose(ql2, qle2[0], rtol=1e-4)
 
-    qler1 = result1.qic()
-    qler2 = result2.qic()
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_gee.py]: does this warn consistently and/or intentionally?  i.e. could be pytest.warns?

File: statsmodels/genmod/tests/test_gee.py [hunk 6]
```diff
@@ -682,6 +687,14 @@ def test_compare_score_test_warnings(self):
             mod = gee.GEE(endog, exog, group)
             _ = mod.compare_score_test(res_sub)
 
+        # Parent and submodel are the same dimension
+        with pytest.warns(UserWarning):
+            w = np.random.uniform(size=n)
+            mod_sub = gee.GEE(endog, exog, group)
+            res_sub = mod_sub.fit()
+            mod = gee.GEE(endog, exog, group)
+            _ = mod.compare_score_test(res_sub)
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_gee.py]: could part of this block be outside of the pytest.warns context?

File: statsmodels/genmod/tests/test_gee.py [hunk 7]
```diff
@@ -635,14 +635,19 @@ def test_compare_score_test(self, cov_struct):
...
+            if call_fit:
+                # Should work with or without fitting the parent model
+                mod.fit()
+            score_results = mod.compare_score_test(res_sub)
+            assert_almost_equal(score_results["statistic"],
+                mod_lr.score_test_results["statistic"])
+            assert_almost_equal(score_results["p-value"],
+                mod_lr.score_test_results["p-value"])
+            assert_almost_equal(score_results["df"],
+                mod_lr.score_test_results["df"])
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_gee.py]: future reference, the lint-compatible version of this is one of:

```
assert_almost_equal(score_results["df"],
                                    mod_lr.score_test_results["df"])

assert_almost_equal(
    score_results["df"],
    mod_lr.score_test_results["df"])


```
